dragonfly/model.py

```diff
@@ -0,0 +1,357 @@
+                passed along to the generated Honeybee Room objects, indicating the
+                simulation will be run once for each unique room and then results
+                will be multiplied. If False, full geometry objects will be written
+                for each and every floor in the building that are represented through
+                multipliers and all resulting multipliers will be 1. Default: True
+            tolerance: The minimum distance in z values of floor_height and
+                floor_to_ceiling_height at which adjacent Faces will be split.
+                If None, no splitting will occur. Default: None.
+        
+        Returns:
+            An array of Honeybee Models that together represent this Dragonfly Model.
+        """
+        # create the model objects
+        if object_per_model is None or object_per_model.title() == 'Building':
+            models = Building.buildings_to_honeybee_self_shade(
+                self._buildings, self._context_shades, shade_distance, use_multiplier,
+                tolerance)
+        elif object_per_model.title() == 'District':
+            models = [Building.buildings_to_honeybee(
+                self._buildings, use_multiplier, tolerance)]
+            for shd in self._context_shades:
+                for model in models:
+                        model.add_shade(shd)
+        else:
+            raise ValueError('Unrecognized object_per_model input: '
+                            '{}'.format(object_per_model))
+        
+        # change the north if the one on this model is not the default
+        if self._north_angle != 0 and self._north_angle != 360:
+            for model in models:
+                model.north_angle = self._north_angle
+        
+        return models
+
+    def to_dict(self, included_prop=None):
+        """Return Model as a dictionary.
+
+        Args:
+            included_prop: List of properties to filter keys that must be included in
+                output dictionary. For example ['energy'] will include 'energy' key if
+                available in properties to_dict. By default all the keys will be
+                included. To exclude all the keys from extensions use an empty list.
+        """
+        base = {'type': 'Model'}
+        base['name'] = self.name
+        base['display_name'] = self.display_name
+        base['properties'] = self.properties.to_dict(included_prop)
+        if self._buildings != []:
+            base['buildings'] = \
+                [bldg.to_dict(True, included_prop) for bldg in self._buildings]
+        if self._context_shades != []:
+            base['context_shades'] = \
+                [shd.to_dict(True, included_prop) for shd in self._context_shades]
+        if self.north_angle != 0:
+            base['north_angle'] = self.north_angle
+
+        return base
+
+    def __add__(self, other):
+        self.add_model(other)
```

Comment:
Nice use of __add__ but I think you need to return something here.

Also you may want to add support for `__iadd__`

```suggestion
        return self.add_model(other)
```

Reply:
That suggested change is just going to return `None` but I see what you mean.  The `__add__` operation should not mutate one of the original objects as it does now but should return a new object that's the sum of the two models.  I'll fix this here and I will send another PR to correct this same situation on the honeybee `Model` object. 

I don't know what adding an integer to a `Model` means so I don't think that I will implement `__iadd__`.

Reply:
> I don't know what adding an integer to a Model means so I don't think that I will implement `__iadd__`.

`__iadd__` will handle something like `model_1 += model_2`. It's not only for integers.

Reply:
Ah, right. Ok, I can put that there.  I guess that's actually what the original code I had for `__add__` should be.

Reply:
In both case you have to return a value. You can basically use one for another. Not sure if you have tests for this but if you add one that can help to ensure it works as expected.

It is a cool feature to have! Not critical for sure.

Reply:
Good point. I should add a `return self` at the end.